### DISCUSSION TITLE
Downloading query result as CSV/JSON

### DISCUSSION THREAD

**Discussion by bstiawan** [Budibase/budibase#4304] [Ideas]
Hi everyone, 

Is there any way we can allow user to download the query result, for example when the query is finished loaded into a table, then we provide option to download as CSV or JSON. Few ways that I can think of:
- Add built in functionality for downloading as file
- Calling a REST that provides the actual file

Thanks!

**Comment by mjashanks** [comment]
Not there at the moment, but we will definitely get to this at some point. Give this a thumbs up / comment: https://github.com/Budibase/budibase/issues/3315 !

**Reply by ttelik** [reply]
What about if an API call returns a file (using the appropriate content-type). Can we send that directly to the user's browser for download?

**Reply by ttelik** [reply]
Having thought a little more about this, I split the API call into two calls. One is called by the REST interface to interact with the data and generates a file (local to API server). The second is accessed as a get request by simply navigating to the link, rather than calling it from the REST connector. I put these two steps sequentially in my action and it works fine.

Open to finding a more elegant solution, but my problem is solved for now.

**Reply by mjashanks** [reply]
Thanks for posting your solution @ttelik 

I don't think there is a more elegant way at the moment. I'd love to add better support for files / attachments all over Budibase. Automations is another place that I'd like to be able to handle files. We'll get there!
